gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -0,0 +1,501 @@
+            error!("Failed to stop LDK Node: {err:?}");
+        } else {
+            info!("LDK Node stopped.");
+        }
+    }
+}
+
+#[async_trait]
+impl ILnRpcClient for GatewayLdkClient {
+    async fn info(&self) -> Result<GetNodeInfoResponse, LightningRpcError> {
+        let node_status = self.node.status();
+
+        let chain_tip_block_summary = self
+            .esplora_client
+            .get_blocks(None)
+            .await
+            .unwrap()
+            .into_iter()
+            .next()
+            .unwrap();
+
+        let esplora_chain_tip_timestamp = chain_tip_block_summary.time.timestamp;
+        let block_height: u32 = chain_tip_block_summary.time.height;
+
+        let synced_to_chain = node_status.latest_wallet_sync_timestamp.unwrap_or_default()
+            > esplora_chain_tip_timestamp
+            && node_status
+                .latest_onchain_wallet_sync_timestamp
+                .unwrap_or_default()
+                > esplora_chain_tip_timestamp;
+
+        Ok(GetNodeInfoResponse {
+            pub_key: self.node.node_id().serialize().to_vec(),
+            // TODO: This is a placeholder. We need to get the actual alias from the LDK node.
+            alias: "LDK Fedimint Gateway Node".to_string(),
+            network: match self.node.config().network {
+                Network::Bitcoin => "main",
+                Network::Testnet => "test",
+                Network::Signet => "signet",
+                Network::Regtest => "regtest",
+                _ => panic!("Unsupported network"),
+            }
+            .to_string(),
+            block_height,
+            synced_to_chain,
+        })
+    }
+
+    async fn routehints(
+        &self,
+        _num_route_hints: usize,
+    ) -> Result<GetRouteHintsResponse, LightningRpcError> {
+        Ok(GetRouteHintsResponse {
+            route_hints: vec![],
+        })
+    }
+
+    async fn pay(
+        &self,
+        invoice: PayInvoiceRequest,
+    ) -> Result<PayInvoiceResponse, LightningRpcError> {
+        let parsed_invoice: Bolt11Invoice =
+            Bolt11Invoice::from_str(&invoice.invoice).map_err(|e| {
+                LightningRpcError::FailedPayment {
+                    failure_reason: format!("Failed to parse invoice: {e:?}"),
+                }
+            })?;
+
+        let payment_id = match self.node.bolt11_payment().send(&parsed_invoice) {
+            Ok(payment_id) => payment_id,
+            Err(e) => {
+                return Err(LightningRpcError::FailedPayment {
+                    failure_reason: format!("LDK payment failed to initialize: {e:?}"),
+                });
+            }
+        };
+
+        // TODO: Find a way to avoid looping/polling to find out when a payment is
+        // completed. `ldk-node` provides `PaymentSuccessful` and `PaymentFailed`
+        // events, but routing them to specific payment IDs isn't straightforward.
+        tokio::time::timeout(Duration::from_secs(30), async {
```

Comment:
nit: use `fedimint_core::runtime::timeout`

Reply:
Can the payment be successful after the 30 second timeout? If so, that could allow an attacker to get paid via lightning and refund their ecash in escrow, stealing money from the gateway

Reply:
Good point... Should we just infinite-loop here and assume that `ldk-node` eventually moves the payment to a non-pending state? (I know this is a pretty gross way of waiting until a payment has been sent, but I believe we need to make some upstream changes to `ldk-node` in order to improve this)

Reply:
Yes I think so. I believe we have similar limitations in LND and CLN currently. I would prefer a payment to be stuck for a long time rather than allowing an attacker to steal funds.

Reply:
Done.